### PR TITLE
Add permissions for operator

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

### CODE REVIEW

**Comment by xuezhaojun** [comment]
The verbs are not very clear right now. Do we need fine-grained verbs in this release? 

/assign @zhujian7 

Feel free to comment, still working on testing.

File: pkg/manager/manifests/permission/role.yaml [hunk 1]
```diff
@@ -10,3 +10,9 @@ rules:
   - apiGroups: ["addon.open-cluster-management.io"]
     resources: ["managedclusteraddons"]
     verbs: ["get", "list", "watch", "update", "patch"]
+  - apiGroups: ["coordination.k8s.io"]
+    resources: ["leases"]
+    verbs: ["*"]
+  - apiGroups: ["apps"]
```
Comment (zhujian7) [inline pkg/manager/manifests/permission/role.yaml:16]: The agent should not operate deployment on the hub side.

Comment (xuezhaojun) [reply pkg/manager/manifests/permission/role.yaml]: Fixed.

**Comment by zhujian7** [comment]
issue reference: https://github.com/stolostron/backlog/issues/21241

**Pull request merged**